edweiss-app/model/quizzes.ts

```diff
@@ -0,0 +1,71 @@
+import { FunctionFolder, FunctionOf } from './functions';
+import { CourseID } from './school/courses';
+
+namespace Quizzes {
+
+    // MCQ: Multiple Choice Question
+    // TF: True or False
+    // "question" is the actual phrase that is asked to the student.
+
+    export interface MCQProposition {
+        description: string;
+        imageUrl?: string;
+        id: number;
+        type: "MCQProposition";
+    }
+
+    export interface MCQ {
+        question: string;
+        imageURL?: string;
+        propositions: MCQProposition[]; // 2 to 8 propositions
+        answersIndices: number[]; // the indices point to the correct propositions
+        type: "MCQ";
+        numberOfAnswers: number;
+    }
+    export interface TF {
+        question: string;
+        imageURL?: string;
+        answer: boolean;
+        type: "TF";
+    }
+
+    export type Exercise = MCQ | TF;
+
+    export interface Quiz {
+        name: string;
+        deadline?: string;
+        exercises: Exercise[];
+        answers: QuizzesAttempts.Answer[];
+    }
+
+    export const Functions = FunctionFolder("action", {
+        createQuiz: FunctionOf<{ quiz: Quizzes.Quiz; courseId: CourseID; }, { id: string; }, "empty_quiz" | "invalid_name" | "not_authorized">("createQuiz"),
+    });
+
+}
+
+export default Quizzes;
+
+export namespace QuizzesAttempts {
+    // export type MCQAnswersIndices = number[]; // unchecked options will not appear in the list in the first place
```

Comment:
The commented-out types and interfaces (like MCQAnswersIndices and TFAnswer) should be either implemented or removed. Leaving unused code can create confusion and clutter in the codebase.

Reply:
Again not my model but thank you